x/wasm/client/cli/tx.go

```diff
@@ -108,9 +116,35 @@ func parseStoreCodeArgs(file string, sender sdk.AccAddress, flags *flag.FlagSet)
 		return types.MsgStoreCode{}, err
 	}
 
+	source, err := flags.GetString(flagSource)
+	if err != nil {
+		return types.MsgStoreCode{}, fmt.Errorf("source: %s", err)
+	}
+	if source == "" {
+		return types.MsgStoreCode{}, fmt.Errorf("source: must not be empty")
+	}
+	builder, err := flags.GetString(flagBuilder)
+	if err != nil {
+		return types.MsgStoreCode{}, fmt.Errorf("builder: %s", err)
+	}
+	if builder == "" {
+		return types.MsgStoreCode{}, fmt.Errorf("builder: must not be empty")
+	}
+	codeHash, err := flags.GetBytesHex(flagCodeHash)
+	if err != nil {
+		return types.MsgStoreCode{}, fmt.Errorf("code-hash: %s", err)
+	}
+	checksum := sha256.Sum256(wasm)
```

Comment:
You are building the hash on the zipped content. I has to be the raw contract instead

Please note that wasmvm is using sha256 but this may change in the future. I have opened https://github.com/CosmWasm/wasmvm/issues/359 to no have this coupling here. Can you link to the issue on this line? IMHO it makes sense to fail fast here